core/bridge_fungible.go

```diff
@@ -0,0 +1,108 @@
+package core
+
+import (
+	"fmt"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/useloopso/BMR/contracts"
+)
+
+func bridgeFungibleTokens(c *LoopsoClient, chain int, transferID [32]byte) {
+	chainInfo := c.chainInfos[chain]
+	client := c.conns[chain]
+
+	srcBridge, err := contracts.NewLoopso(common.HexToAddress(chainInfo.BridgeAddress), client)
+	if err != nil {
+		fmt.Println("failed to init bridge contract: ", err)
+		return
+	}
+
+	tokenTransfer, err := srcBridge.TokenTransfers(nil, transferID)
+	if err != nil {
+		fmt.Println(err)
+		return
+	}
+
+	dstChainId := int(tokenTransfer.TokenTransfer.DstChain.Int64())
+	dstBridgeAddress := c.chainInfos[dstChainId].BridgeAddress
+	dstBridgeClient := c.conns[dstChainId]
+
+	dstBridge, err := contracts.NewLoopso(common.HexToAddress(dstBridgeAddress), dstBridgeClient)
+	if err != nil {
+		fmt.Println("failed to init bridge: ", err)
+		return
+	}
+
+	// check whether token is supported already
+	isTokenSupported, err := dstBridge.IsTokenSupported(nil, tokenTransfer.TokenTransfer.TokenAddress, big.NewInt(int64(chain)))
+	if err != nil {
+		fmt.Println("failed to check is token supported:", err)
+		return
+	}
+
+	auth, err := c.Auth(dstChainId)
+	if err != nil {
+		fmt.Println(err)
+		return
+	}
+
+	if !isTokenSupported {
+		// if no --> all attestToken on destination chain
+		att := attestationFromTokenTransfer(tokenTransfer, client)
+		tx, err := dstBridge.AttestToken(auth, att)
+		if err != nil {
+			fmt.Println("failed to attest token: ", err)
+			return
+		}
+		fmt.Println("attest token tx hash: ", tx.Hash())
+	}
```

Comment:
This code minted an attestation that didn't work when I ran it myself on a new ERC20/ERC721, not sure why... can you take a look @lajosdeme ?

Reply:
it failed to use that attestation to mint new tokens on the destChain

Reply:
yeah, the issue is that we don't wait for the `dstBridge.AttestToken` to be confirmed, and try to bridge before the wrapped token is even deployed. Pushing a fix for this soon

Reply:
fixed here: 88257bf7a9581265c37d3e620bf772987f4bea6f